sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,422 @@ def to_rotation_matrix(self, v=None):
+
+        Returns
+        =======
+
+        Quaternion: representing axis of the provided quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.axis()
+        0 + sqrt(3)/3*i + sqrt(3)/3*j + sqrt(3)/3*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.axis()
+        0 + 8*sqrt(377)/377*i + sqrt(377)/29*j + 12*sqrt(377)/377*k
+
+        See Also
+        ========
+
+        vector_part
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d).normalize()
+
+    def is_pure(self):
+        """
+        Returns true if the quaternion is pure, false if the quaternion is not pure
+        or returns none if it is unknown.
+
+        Explanation
+        ===========
+
+        A pure quaternion (also a vector quaternion) is a quaternion with scalar
+        part equal to 0.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.is_pure()
+        False
+
+        >>> q = Quaternion(0, 8, 13, 12)
+        >>> q.is_pure()
+        True
+
+        See Also
+        ========
+        scalar_part
+
+        """
+
+        return self.a.is_zero
+
+    def is_zero_quaternion(self):
+        """
+        Returns true if the quaternion is zero or false if it is not zero and None
+        if the value is unknown.
+
+        Explanation
+        ===========
+
+        A zero quaternion is a quaternion with it's scalar part and vector part
+        both equal to 0.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 0, 0, 0)
+        >>> q.is_zero_quaternion()
+        False
+
+        >>> q = Quaternion(0, 0, 0, 0)
+        >>> q.is_zero_quaternion()
+        True
+
+        See Also
+        ========
+        scalar_part
+        vector_part
+
+        """
+
+        return self.norm().is_zero
+
+    def angle(self):
+        r"""
+        Returns the angle of the quaternion measured in the real-axis plane.
+
+        Explanation
+        ===========
+
+        If q is a quaternion given by q = a + b*i + c*j + d*k where a, b, c and d
+        are real numbers then the angle of the quaternion is given by
+
+            angle := atan2 (\sqrt(b^2+c^2+d^2), a)
+
+        Returns
+        =======
+
+        expr: The angle of the quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 4, 4, 4)
+        >>> q.angle()
+        atan(4*sqrt(3))
+
+        >>> q = Quaternion(0, 8, 13, 12)
+        >>> q.angle()
+        pi/2
+
+        """
+
+        return atan2(self.vector_part().norm(), self.scalar_part())
+
+    @classmethod
+    def coplanar(cls, q1, q2):
+        """
+        Returns True if the plane of the two quaternions are coplanar else returns False.
```

Comment:
I'm not sure I understand what this means. Should it be plural "planes"?

What does it mean for planes to be "coplanar" anyway?

I guess this can be worded better.